starlink/dish.py

```diff
@@ -8,430 +8,241 @@
-        "sinr",
-        "popPingLatencyMs",
-        "downlinkThroughputBps",
-        "uplinkThroughputBps",
-        "tiltAngleDeg",
-        "boresightAzimuthDeg",
-        "boresightElevationDeg",
-        "attitudeEstimationState",
-        "attitudeUncertaintyDeg",
-        "desiredBoresightAzimuthDeg",
-        "desiredBoresightElevationDeg",
-    ]
-    mobile_columns = [
-        "latitude",
-        "longitude",
-        "altitude",
-        "qScalar",
-        "qX",
-        "qY",
-        "qZ",
-    ]
-    header = base_columns + (mobile_columns if mobile else [])
-    csv_writer.writerow(header)
-
-
-def extract_status_fields(status: dict) -> list:
-    alignment = status.get("alignmentStats", {})
-    return [
-        time.time(),
-        status.get("phyRxBeamSnrAvg", 0),
-        status.get("popPingLatencyMs", 0),
-        status.get("downlinkThroughputBps", 0),
-        status.get("uplinkThroughputBps", 0),
-        alignment.get("tiltAngleDeg", 0),
-        alignment.get("boresightAzimuthDeg", 0),
-        alignment.get("boresightElevationDeg", 0),
-        alignment.get("attitudeEstimationState", ""),
-        alignment.get("attitudeUncertaintyDeg", 0),
-        alignment.get("desiredBoresightAzimuthDeg", 0),
-        alignment.get("desiredBoresightElevationDeg", 0),
-    ]
-
-
-def extract_location_fields(location_data: dict, status: dict) -> list:
-    lla = location_data.get("getLocation", {}).get("lla", {})
-    quaternion = status.get("ned2dishQuaternion", {})
-    return [
-        lla.get("lat", 0),
-        lla.get("lon", 0),
-        lla.get("alt", 0),
-        quaternion.get("qScalar", 0),
-        quaternion.get("qX", 0),
-        quaternion.get("qY", 0),
-        quaternion.get("qZ", 0),
-    ]
-
-
-def get_sinr() -> None:
+    logger.info(f"{name}, {threading.current_thread()}")
+
+    filename = f"{GRPC_DATA_DIR}/{ensure_data_directory(GRPC_DATA_DIR)}/GetStatus-{date_time_string()}.txt"
+    
+    # Create and execute GRPC command to get status
+    cmd = GrpcCommand("status", '{"get_status":{}}')
+    cmd.save_to_file(filename)
+
+def monitor_dish_state() -> None:
+    """Collect SINR and location data over time."""
     name = "GRPC_phyRxBeamSnrAvg"
     logger.info(f"{name}, {threading.current_thread()}")
 
-    FILENAME = "{}/{}/GRPC_STATUS-{}.csv".format(
-        GRPC_DATA_DIR, ensure_data_directory(GRPC_DATA_DIR), date_time_string()
-    )
-
-    status_cmd = [
-        "grpcurl",
-        "-plaintext",
-        "-d",
-        '{"get_status":{}}',
-        STARLINK_GRPC_ADDR_PORT,
-        "SpaceX.API.Device.Device/Handle",
-    ]
-    location_cmd = [
-        "grpcurl",
-        "-plaintext",
-        "-d",
-        '{"get_location":{}}',
-        STARLINK_GRPC_ADDR_PORT,
-        "SpaceX.API.Device.Device/Handle",
-    ]
-
-    with open(FILENAME, "w", newline="") as outfile:
+    filename = f"{GRPC_DATA_DIR}/{ensure_data_directory(GRPC_DATA_DIR)}/GRPC_STATUS-{date_time_string()}.csv"
```

Comment:
now I tend to split location data and the existing `GRPC_STATUS_*.csv` into two different files with same experiment id as indicated in the filenames.

Current `GRPC_STATUS_*.csv` includes

```
timestamp,sinr,popPingLatencyMs,downlinkThroughputBps,uplinkThroughputBps,tiltAngleDeg,boresightAzimuthDeg,boresightElevationDeg,attitudeEstimationState,attitudeUncertaintyDeg,desiredBoresightAzimuthDeg,desiredBoresightElevationDeg
```

We can add `quaternion_qScalar, quaternion_qX, quaternion_qY, quaternion_qZ` to `GRPC_STATUS_*.csv`, but move location related to another file:

```
gps_time, utc_time, lat, lon, alt, uncertainty_valid, uncertainty
```

which also splits the mobile mode and the stationary mode, which might be easier for post-processing as well?